File: dotnet/src/Microsoft.Agents.Protocol.Validation/Validators/ContentValidators/TypingIndicatorContentValidator.cs
using Microsoft.Agents;

namespace Microsoft.Agents.Protocol.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for TypingIndicatorContent.
/// </summary>
public class TypingIndicatorContentValidator : ContentValidatorBase<TypingIndicatorContent>
{
    public override ValidationResult Validate(TypingIndicatorContent content, ValidationContext? context = null)
    {
        // TODO: Implement specific validation rules for TypingIndicatorContent
        return ValidationResult.Success();
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/TypingIndicatorContentValidator.ts
import type { TypingIndicatorContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';

/**
 * Validator for TypingIndicatorContent.
 */
export class TypingIndicatorContentValidator extends ContentValidatorBase<TypingIndicatorContent> {
  public validate(content: TypingIndicatorContent, context?: ValidationContext): ValidationResult {
    // TypingIndicatorContent has no strict validation requirements
    return ValidationResult.success();
  }
}
